skore/src/skore/sklearn/_plot/metrics/roc_curve.py

```diff
@@ -0,0 +1,865 @@
+            classification, the dictionary has a single key for the positive class. For
+            multiclass, there is one key per class.
+
+        pos_label : int, float, bool, str or None
+            The positive class label for binary classification.
+            Must be provided for binary classification.
+            Ignored for multiclass.
+
+        data_source : {"train", "test", "X_y"}
+            The data source used to compute the ROC curve.
+
+        ml_task : {"binary-classification", "multiclass-classification"}
+            The machine learning task.
+
+        ax : matplotlib.axes.Axes
+            The axes on which to plot.
+
+        estimator_names : list of str
+            The names of the estimators.
+
+        roc_curve_kwargs : list of dict
+            List of dictionaries containing keyword arguments to customize the ROC
+            curves. The length of the list should match the number of curves to plot.
+
+        Returns
+        -------
+        ax : matplotlib.axes.Axes
+            The axes with the ROC curves plotted.
+
+        lines : list of matplotlib.lines.Line2D
+            The plotted ROC curve lines.
+
+        info_pos_label : str or None
+            String containing positive label information for binary classification,
+            None for multiclass.
+        """
+        lines: list[Line2D] = []
+        line_kwargs: dict[str, Any] = {}
+
+        if ml_task == "binary-classification":
+            assert pos_label is not None, (
+                "pos_label should not be None with binary classification."
+            )
+            for est_idx, est_name in enumerate(estimator_names):
+                fpr_est = fpr[pos_label][est_idx]
+                tpr_est = tpr[pos_label][est_idx]
+                roc_auc_est = roc_auc[pos_label][est_idx]
+
+                line_kwargs_validated = _validate_style_kwargs(
+                    line_kwargs, roc_curve_kwargs[est_idx]
+                )
+                line_kwargs_validated["label"] = (
+                    f"{est_name} - {data_source} set (AUC = {roc_auc_est:0.2f})"
+                )
+                (line,) = ax.plot(fpr_est, tpr_est, **line_kwargs_validated)
+                lines.append(line)
+
+            info_pos_label = (
+                f"\n(Positive label: {pos_label})" if pos_label is not None else ""
+            )
+        else:  # multiclass-classification
+            info_pos_label = None  # irrelevant for multiclass
+            class_colors = sample_mpl_colormap(
+                colormaps.get_cmap("tab10"), 10 if len(fpr) < 10 else len(fpr)
+            )
+
+            for est_idx, est_name in enumerate(estimator_names):
+                est_color = class_colors[est_idx]
+
+                for class_idx, class_ in enumerate(fpr):
+                    fpr_est_class = fpr[class_][est_idx]
+                    tpr_est_class = tpr[class_][est_idx]
+                    roc_auc_mean = np.mean(roc_auc[class_][est_idx])
```

Comment:
@chkoar

Previously we had:

```python
roc_auc_mean = np.mean(roc_auc[class_])
```

So we averaged the statistics of all estimators for a class and not for a specific estimator.

Reply:
Now it correctly displays a separate value for each curve.

However, in my [example](https://github.com/probabl-ai/skore/pull/1361#1443), it is not particularly useful because I have two classifiers for ten classes, resulting in 20 curves. It would be great to have a built-in option to view the micro or macro-averaged curve for each classifier, using either the `OvR` or `OvO` decomposition. If you agree I could open a separate issue.

Reply:
I think that the first thing that I had in head to solve this problem (and some other plot) is to break down the plot for cases where we know we have too much information: https://github.com/probabl-ai/skore/issues/1445

Once done, I think that I would prefer to implement the OvO approach but not necessarly rush for the averaging because sometimes people don't understand what average they are looking at and forcing them to see all the curves is a better default in this regards. But it needs to be readable :)